Userland/Utilities/uniq.cpp

```diff
@@ -5,75 +5,127 @@
  */
 
 #include <AK/RefPtr.h>
+#include <AK/StringView.h>
 #include <LibCore/ArgsParser.h>
+#include <LibCore/Stream.h>
 #include <LibCore/System.h>
-#include <errno.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
 #include <unistd.h>
 
-struct linebuf {
-    char* buf = NULL;
-    size_t len = 0;
-};
+static Optional<String> get_line_content(StringView line, size_t count, bool show_duplicates, bool print_count)
+{
+    if (!show_duplicates || count > 0) {
+        if (print_count)
+            return String::formatted("{} {}\n", count + 1, line);
+        else
+            return String::formatted("{}\n", line);
+    }
+    return {};
+}
 
-static FILE* get_stream(char const* filepath, char const* perms)
+static StringView skip(StringView buf, unsigned int nchars, unsigned int nfields)
 {
-    FILE* ret;
+    StringView tmp(buf);
```

Comment:
There's no need to copy the string view here, just use the parameter directly (also `tmp` is  arguably not a good name).